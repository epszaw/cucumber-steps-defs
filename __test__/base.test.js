const { vol } = require('memfs')
const { readFileSync } = require('fs')
const path = require('path')
const {
  createTree,
  createDefsFromTree,
  createDefsFromFeatureSource,
  convert,
} = require('index')

describe('base', () => {
  const treeWithoutStrings = require('./fixtures/trees/withoutStrings.json')
  const treeWithStrings = require('./fixtures/trees/withStrings.json')
  const featureWithoutStrings = readFileSync(
    path.join(__dirname, './fixtures/features/withoutStrings.feature'),
    'utf8',
  )
  const featureWithStrings = readFileSync(
    path.join(__dirname, './fixtures/features/withStrings.feature'),
    'utf8',
  )

  beforeEach(() => {
    vol.reset()
    vol.fromJSON({
      'features/withoutStrings.feature': featureWithoutStrings,
      'features/withStrings.feature': featureWithStrings,
      defs: {},
    })
  })

  describe('createTree', () => {
    it('should create tree from feature sources', () => {
      expect(createTree(featureWithoutStrings)).toEqual(treeWithoutStrings)
      expect(createTree(featureWithStrings)).toEqual(treeWithStrings)
    })
  })

  describe('createDefsFromTree', () => {
    it('should correctly transform prepared steps tree', () => {
      expect(createDefsFromTree(treeWithoutStrings)).toMatchSnapshot()
      expect(createDefsFromTree(treeWithStrings)).toMatchSnapshot()
    })
  })

  describe('createDefsFromFeatureSource', () => {
    it('should correctly transform features sources to steps defs', () => {
      expect(
        createDefsFromFeatureSource(featureWithoutStrings),
      ).toMatchSnapshot()
      expect(createDefsFromFeatureSource(featureWithStrings)).toMatchSnapshot()
    })
  })

  describe('convert', () => {
    it('should do something', async () => {
      expect.assertions(2)

      await convert('features/withStrings.feature', 'defs')
      await convert('features/withoutStrings.feature', 'defs')

      expect(
        vol.readFileSync('defs/withoutStrings.js', 'utf8'),
      ).toMatchSnapshot()
      expect(vol.readFileSync('defs/withStrings.js', 'utf8')).toMatchSnapshot()
    })
  })
})
