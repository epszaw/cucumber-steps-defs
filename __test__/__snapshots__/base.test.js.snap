// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base convert should do something 1`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition', function() {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome', function() {
  return true
})"
`;

exports[`base convert should do something 2`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition with {string} and {string}', function(param1, param2) {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome equals {string}', function(param1) {
  return true
})"
`;

exports[`base createDefsFromFeatureSource should correctly transform features sources to steps defs 1`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition', function() {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome', function() {
  return true
})"
`;

exports[`base createDefsFromFeatureSource should correctly transform features sources to steps defs 2`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition with {string} and {string}', function(param1, param2) {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome equals {string}', function(param1) {
  return true
})"
`;

exports[`base createDefsFromTree should correctly transform prepared steps tree 1`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition', function() {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome', function() {
  return true
})"
`;

exports[`base createDefsFromTree should correctly transform prepared steps tree 2`] = `
"const { Given, When, And, Then } = require('cucumber')

Given('precondition with {string} and {string}', function(param1, param2) {
  return true
})

When('action', function() {
  return true
})

And('addition', function() {
  return true
})

Then('testable outcome equals {string}', function(param1) {
  return true
})"
`;
